Feature/fe dashboard

vic2019 opened the pull request.

Commit:
Create Dashboard page

Commit:
Edit style for Dashboard

Commit:
Resolve conflict to merge dev into feature/FE-dashboard

Commit:
Adjust font; add profile button and avatar

Commit:
Add badge to notification

Commit:
Work on add url bar

Commit:
Edit style for AddItem

maismin commented on client/src/pages/Dashboard.js:
```diff
@@ -0,0 +1,18 @@
+import React, { useState, useEffect } from 'react';
+import ToolBar from '@material-ui/core/ToolBar';
```
ToolBar should be Toolbar

maismin commented on client/src/pages/Dashboard.js:
```diff
@@ -0,0 +1,18 @@
+export default function Dashboard(props) {
+  return (
+    <div id="Dashboard" style={dashboardStyle}>
+      <ToolBar />
```
ToolBar should be Toolbar

maismin reviewed (changes requested):
- Visually, the AppBar looks very good. Intuitively, I think the image logo should redirect the user to the '/' page.
- I noticed that the box-shadow attribute was used in many places. Perhaps that should be moved to the themes.js file. PR #25 has the box-shadow settings in theme.js.
- I noticed that a few files did not go through eslint and prettier. When I pulled your PR, I could not run the app because of a misspelling. If you have trouble setting up eslint and prettier to work automatically on vscode, lets meet and get you started on them.

aeciorc commented on client/src/themes/theme.js:
```diff
@@ -1,4 +1,5 @@
 import { createMuiTheme } from '@material-ui/core';
+import red from '@material-ui/core/colors/red';
```
Is this unused?

aeciorc commented on client/src/components/AddItem.js:
```diff
@@ -0,0 +1,115 @@
+          >
+            <option value='default'>Select list</option>
+            {options.map(list => (
+              <option value="list.title">{list.title}</option>
```
```suggestion
              <option value={list.title}>{list.title}</option>
```

aeciorc commented on client/src/components/AddItem.js:
```diff
@@ -0,0 +1,115 @@
+            className={classes.select}
+          >
+            <option value='default'>Select list</option>
+            {options.map(list => (
```
remember the `key` prop when you map

aeciorc commented on client/src/components/AddItem.js:
```diff
@@ -0,0 +1,115 @@
+          placeholder='Paste your link here'
+        />
+        <span className={classes.selectContainer}>
+          <select
```
Consider using mui Select, Input (with disableUnderline) and MenuItem instead of select, input and option.

Commit:
.

Esodey commented on client/src/components/AddItem.js:
```diff
@@ -0,0 +1,115 @@
+  const [input, set] = useState({ text: '', select: '' });
+  const [options, setOptions] = useState([]);
+
+  const handleInputChange = (e) => {
```
Could potentially make this into one function?

Commit:
Merge branch 'dev' into feature/FE-dashboard

Commit:
Fix styling for AddItem. Integrate CreateList

maismin committed:
Fix theme file

- Remove global settings for MuiInput
- Fix CreateListForm to update for the global changes in theme.js

maismin commented on client/src/components/Dashboard/AddItem.js:
```diff
@@ -0,0 +1,135 @@
+  palette: {
+    primary: { main: '#DF1B1B' },
+  },
+});
```
Since the primary palette is used throughout the entire project, it's best to move this setting into the themes/theme.js file. The theme.js already has this color set. Components can access the global themes by passing props into the useStyles() function.

maismin commented on client/src/components/Dashboard/AddItem.js:
```diff
@@ -0,0 +1,135 @@
+}));
+
+export default function AddItem() {
+  const classes = useStyles();
```
Changing AddItem() to AddItem(props) and useStyles() to useStyles(props) provides it with the theme object.

maismin commented on client/src/components/Dashboard/AddItem.js:
```diff
@@ -0,0 +1,135 @@
+
+  const handleSelectChange = e => {
+    set({ select: e.target.value });
+  };
```
Updating state via useState and setState are different when it comes to objects
With setState:

- Passing an object argument will update only the properties in said object

With useState:

- The object argument replaces the previous state entirely

So the eventHandlers on lines 79 and 83 will update text and select properties respectively, but throw out anything else not in the object argument. This is not what we want. A better way might be to use useState() to keep track of a single value instead of objects. Here's a [link](https://blog.logrocket.com/a-guide-to-usestate-in-react-ecb9952e406c/) that might clear some confusion regarding useState.

The naming convention with useState is
``
const [varname, setVarname] = useState('')
``

The eventHandlers can be passed as anonymous functions since they are simple.

maismin commented on client/src/components/Dashboard/AddItem.js:
```diff
@@ -0,0 +1,135 @@
+import Input from '@material-ui/core/Input';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import Typography from '@material-ui/core/Typography';
```
Can be rewritten as
``
import {
  Button,
  Input,
  MenuItem,
  Select,
  Typography,
} from '@material-ui/core';
``
This has less code and makes it easier to see that all these functions are from the @material-ui/core module.

maismin commented on client/src/components/Dashboard/AddItem.js:
```diff
@@ -0,0 +1,135 @@
+            className={classes.addButton}
+          >
+            Add
+          </Button>
```
I created the ThemeButton component specifically for this use case.

maismin committed:
Refactor AddItem component

- Rename AddItem to AddItemForm because the website uses two forms: the
other one is a form within a dialog component

- Replace Button component with our ThemeButton component

- Replace span tag with a material-ui Box component to include the
Input, Select, and ThemeButton components and style the rounded
rectangular shape

- Remove the useEffect hook as the data will be passed as props from the
redux store

- Add a divider component between the Input and Select component

- Refactor mapping shopping lists into MenuItem components into its own
function for clarity

- Refactor code to reduce redundancy and follow best practices

maismin committed:
Refactor TopNavBar

- Refactor imports to reduce redundancy

maismin committed:
Fix AddItemForm component

- Made changes so that it is more responsive

- More changes will be needed to make it more responsive to specific
breakpoints

maismin committed:
Refactor ShoppingLists component

- Refactor imports section to remove redundancy

- Add comments to indicate mock data or areas to be improved on later

- Remove unnecessary code comments

- Refactor mapping shoppinglists to Card components into its own
function for clarity

- Remove hooks from the component as the data will be retrived from the
redux store

Commit:
Edit padding and height for AddItem; edit image height for ShoppingList
